### PR TITLE
Allow status bar's time/stats widgets to show independently

### LINKED ISSUE #504: split Level Stats and Time widget into two separate choices

### PR DESCRIPTION
**Pull request by facespkz**
Fixes #504

### CODE REVIEW

**Comment by NightFright2k19** [comment]
Thanks a lot for doing this, guys!

**Comment by MrAlaux** [comment]
What if instead of using two new variables, `hud_timests` were allowed to go up to 3? This would imply turning the menu item into an S_CHOICE type too, which wouldn't take as much space. Furthermore, it wouldn't need that conversion part for older CFGs, as long as `hud_timests = 1` is used to display both lines.

**Comment by facespkz** [comment]
I do/did have a stash for that, just without the displaying both lines if equal to 1. I decided upon this since it would be easier to read and wouldn't need to convert to S_CHOICE, though I guess that's what enums are for.

**Comment by fabiangreffrath** [comment]
Thank you for your effort! Sorry, but there is no way I can accept three menu switches and variables to handle display of two widgets. Please follow the S_CHOICE approach and ignore config transition for now.

**Comment by facespkz** [comment]
Sure thing, I'll see what I can do!

**Comment by facespkz** [comment]
Turns out that having 'both' option be 1 instead of 3 also made this method a bit easier to understand, since you only need modulo to determine what widgets can be drawn. Thanks @MrAlaux

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]
Much better now! Though I'd really like to avoid using modulo. Please change the menu choice to `"OFF", "TIME", "STATS", "BOTH"` and use binary-and accordingly.

Yes, this means that users will have to adapt their config *once*, but we don't have to carry the code that tries to prevent this forever.

File: Source/hu_stuff.c [hunk 1]
```diff
@@ -1088,11 +1090,10 @@ void HU_Drawer(void)
       if (crispy_hud)
       {
           ST_Drawer (false, true);
-          if (hud_timests)
-          {
+          if (1 & hud_timests)
```
Comment (fabiangreffrath) [inline Source/hu_stuff.c:1093]: Sorry for nit-picking, but could you reverse this, please? Like `hud_timests & 1`. Bonus points for introducing an `enum` to hold thd calues, e.g. `TIMESTS_TIME =1`.

Comment (facespkz) [reply Source/hu_stuff.c:1093]: Wasn't too sure where to put these enums, so I put them right beneath the normal HUD stuff

**Review by fabiangreffrath — APPROVED** [review]
LGTM. What do you say @rfomin ?

**Review by rfomin — APPROVED** [review]
Looks good!

**Comment by fabiangreffrath** [comment]
Thank you!

**Pull request merged**